operators/src/pro/cache/tile_cache.rs

```diff
@@ -100,9 +100,42 @@ pub struct OperatorTileCache {
 
 identifier!(QueryId);
 
+struct FindMatchResult {
+    match_found: Option<CacheEntryId>,
+    expired_entries: Vec<CacheEntryId>,
+}
+
 impl OperatorTileCache {
-    pub fn find_match(&self, query: &RasterQueryRectangle) -> Option<(&CacheEntryId, &CacheEntry)> {
-        self.entries.iter().find(|(_id, r)| r.matches(query))
+    /// Find a match for the query in the cache.
+    ///
+    /// This method will also remove expired entries it encounters.
+    ///
+    /// It returns the expired entries, s.t. the caller can remove them from the lru index structure.
+    fn find_match(&mut self, query: &RasterQueryRectangle) -> FindMatchResult {
+        let mut expired_entries = vec![];
+
+        let match_found = self.entries.iter().find(|(id, r)| {
+            let matches = r.matches(query);
+
+            if matches && r.is_expired() {
+                expired_entries.push(**id);
+
+                return false;
+            }
+
+            matches
+        });
+
+        let match_found = match_found.map(|(id, _)| *id);
```

Comment:
just do the map there?

Reply:
where?

Reply:
126

Reply:
The closure in 126 returns a bool and not a CacheEntryId I could put it in 127 but then rustfmt breaks the stuff down even further and we end up in line 131 instead of 129, so haven't really saved anything.